From ce76cad36e8011e1fc052ae00e03228cd4e0b51d Mon Sep 17 00:00:00 2001
From: Glen Huang <me@glenhuang.com>
Date: Tue, 21 Nov 2023 11:07:07 +0800
Subject: [PATCH] strongswan: add empty config

Without it, using uci to manipulate ipsec config can result in errors,
making it much difficult to use in uci-defaults for example.

Signed-off-by: Glen Huang <me@glenhuang.com>
---
 net/strongswan/Makefile           | 4 +++-
 net/strongswan/files/ipsec.config | 2 ++
 2 files changed, 5 insertions(+), 1 deletion(-)
 create mode 100644 net/strongswan/files/ipsec.config

diff --git a/feeds/packages/net/strongswan/Makefile b/feeds/packages/net/strongswan/Makefile
index 20790d8ea75ce..7b67939399b12 100644
--- a/feeds/packages/net/strongswan/Makefile
+++ b/feeds/packages/net/strongswan/Makefile
@@ -325,7 +325,7 @@ endef
 define Package/strongswan-isakmp/description
 $(call Package/strongswan/description/Default)
  This meta-package contains only dependencies to establish  ISAKMP /
- IKE PSK connections, dropping other capabilities in favor of small size 
+ IKE PSK connections, dropping other capabilities in favor of small size
  Can fit most routers even with 4Mb flash (after removing IPv6 support).
 endef
 
@@ -585,6 +585,8 @@ define Package/strongswan-swanctl/install
 	$(INSTALL_DIR) $(1)/usr/sbin
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/swanctl $(1)/usr/sbin/
 	$(INSTALL_BIN) ./files/swanctl.init $(1)/etc/init.d/swanctl
+	$(INSTALL_DIR) $(1)/etc/config
+	$(INSTALL_CONF) ./files/ipsec.config $(1)/etc/config/ipsec
 endef
 
 define Package/strongswan-gencerts/install
diff --git a/feeds/packages/net/strongswan/files/ipsec.config b/feeds/packages/net/strongswan/files/ipsec.config
new file mode 100644
index 0000000000000..4e976063f80d0
--- /dev/null
+++ b/feeds/packages/net/strongswan/files/ipsec.config
@@ -0,0 +1,2 @@
+# For strongSwan ipsec config documentation see
+# https://openwrt.org/docs/guide-user/services/vpn/strongswan/start
